pkg/openshift/certgen/config_test.go

```diff
@@ -0,0 +1,116 @@
+package certgen
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/Azure/acs-engine/pkg/openshift/filesystem"
+)
+
+type fakefilesystem struct {
+}
+
+func (fakefilesystem) WriteFile(filename string, data []byte, perm os.FileMode) error {
+	// check that you've been asked to write the given file with sane permissions & owner
+	fmt.Printf("Filename: %v	Permissions: %04o   string: %v\n", filename, perm.Perm(), perm.String())
+	return nil
+}
+
+func (fakefilesystem) Close() error {
+	return nil
+}
+
+func TestConfigFilePermissions(t *testing.T) {
+	c := Config{
+		Templateinterface: ActualTemplates{},
+		Master: &Master{
+			Hostname: fmt.Sprintf("%s-master-%s-0", "test", "test"),
+			IPs: []net.IP{
+				net.ParseIP("10.0.0.1"),
+			},
+			Port: 8443,
+		},
+		ExternalMasterHostname: "testhostname.cloudapp.azure.com",
+		ClusterUsername:        "test",
+		ClusterPassword:        "test",
+		AzureConfig: AzureConfig{
+			TenantID:        "tenantid",
+			SubscriptionID:  "subid",
+			AADClientID:     "clientid",
+			AADClientSecret: "secret",
+			ResourceGroup:   "resourcegroup",
+			Location:        "location",
+		},
+	}
+
+	err := c.PrepareMasterCerts()
```

Comment:
because you're using faketemplates and that will cover the cases for all the different files you output, you don't need any of this - just do WriteMasterFiles